cmd/controller.go

```diff
@@ -101,15 +117,37 @@ func newControllerCommand() *cobra.Command {
 			ctrl := controller.NewController(k8sClient, dynamicClient, argocdService, namespace, appLabelSelector, registry)
 			err = ctrl.Init(context.Background())
 			if err != nil {
-				return err
+				panic(err)
+			}
+
+			if leaderElectionConfiguration.LeaderElect {
+				leaderElector, err := k8s.CreateLeaderElector(
+					leaderElectionLockName,
+					namespace,
+					leaderElectionId,
+					k8sClient,
+					leaderElectionConfiguration,
+					func(ctx context.Context) {
+						ctrl.Run(ctx, processorsCount)
+					})
+				if err != nil {
+					return err
+				}
+
+				go leaderElector.Run(context.Background())
+				<-context.Background().Done()
+			} else {
+				go ctrl.Run(context.Background(), processorsCount)
+				<-context.Background().Done()
```

Comment:
I think we can make it common for line of 138 and 141.

Reply:
I'm not sure how to do it. I guess i can't assign 137 or 140 line expression to some var. Can you suggest how to do it?